lldb/source/Plugins/ExpressionParser/Swift/SwiftUserExpression.cpp

```diff
@@ -255,12 +247,18 @@ void SwiftUserExpression::ScanContext(ExecutionContext &exe_ctx, Status &err) {
     m_is_class |= info.type_flags.Test(lldb::eTypeIsClass);
 
   // Handle weak self.
-  auto *ref_type =
-      llvm::dyn_cast_or_null<swift::ReferenceStorageType>(info.swift_type);
-  if (ref_type && ref_type->getOwnership() == swift::ReferenceOwnership::Weak) {
-    m_is_class = true;
-    m_is_weak_self = true;
-  }
+
+  auto ts = info.type.GetTypeSystem().dyn_cast_or_null<TypeSystemSwift>();
+  if (!ts)
+    return;
+
+  if (auto ownership_kind = ts->GetNonTriviallyManagedReferenceKind(
+          info.type.GetOpaqueQualType()))
+    if (*ownership_kind ==
+        SwiftASTContext::NonTriviallyManagedReferenceKind::eWeak) {
+      m_is_class = true;
+      m_is_weak_self = true;
+    }
```

Comment:
why did this code have to change?

Reply:
Because it operated on a `swift::Type` rather than a CompilerType.